core/variant/variant_call.cpp

```diff
@@ -753,40 +753,56 @@ struct _VariantCall {
 	static PackedInt32Array func_PackedByteArray_decode_s32_array(PackedByteArray *p_instance) {
 		uint64_t size = p_instance->size();
 		PackedInt32Array dest;
-		ERR_FAIL_COND_V_MSG(size < sizeof(int32_t), dest, "Size didn't match array of size int32_t, maybe you are trying to convert to the wrong type?");
+		if (size == 0) {
+			return dest;
+		}
+		static_assert(sizeof(dest[0]) == sizeof(int32_t));
+		ERR_FAIL_COND_V_MSG(size % sizeof(int32_t), dest, "Size didn't match array of size int32_t, maybe you are trying to convert to the wrong type?");
 		const uint8_t *r = p_instance->ptr();
 		dest.resize(size / sizeof(int32_t));
-		memcpy(dest.ptrw(), r, size);
+		memcpy(dest.ptrw(), r, dest.size() * sizeof(int32_t));
```

Comment:
This is equivalent to `size` unless we failed to the requested size above. Can that happen and still make a meaningful copy?

Reply:
Resizing can fail, it returns an `Error` which could be checked. Since here `dest` is being enlarged I think resizing can only fail in out of memory case. As far as I've checked `resize` will either succeed and resize to the given size or it will fail and leave the previous size unchanged. So here if `dest.resize()` will fail then it would produce an error and `dest.size()` will remain 0. Meaning if `size` was used it would try to `memcpy` to the memory being out of bounds of `dest`'s allocated memory. Using `dest.size()` prevents this, it's a good change.
I think the question is whether the error of `dest.resize()` should be checked explicitly instead (before `memcpy` is called).

Reply:
Good point. I hadn't considered that resize could fail: I just wrote it that way to be defensive. I added another check:
```
ERR_FAIL_COND_V (dest.size() == 0, dest); // Avoid UB in case resize failed.
```
to handle the case that resize fails, since memcpy 0 is undefined behavior.